File: lab3-primit/java/BoolChar.java
import java.util.Scanner;
import java.util.InputMismatchException;

public class BoolChar
{
    private static int getNumber()
    {
        boolean haveResult = false;
        int result = 0;
        Scanner stdin = new Scanner(System.in);

        while ( ! haveResult )
        {
            System.out.print("Input number: ");
            try
            {
                result = stdin.nextInt();
                haveResult = true;
            }
            catch (InputMismatchException e)
            {
                System.out.println("Not a valid number.");
                stdin.nextLine(); // skip the invalid input
            }
        }

        return result;
    }

    public static void main(String[] args)
    {

        String aS = "a";
        System.out.println(aS + " contains " + aS.toCharArray().length + " chars");
        int aI = Character.codePointAt(aS.toCharArray(), 0);
        System.out.println("code point of " + aS + " is " + aI);

        String cleff = "𝄞";
        System.out.println(cleff + " contains " + cleff.toCharArray().length + " chars");

        int n = getNumber();

        while ( n == 0 )
        {
            System.out.println("Please, input a non-zero number.");
            n = getNumber();
        }

        System.out.println("You typed: " + n);
    }
}

File: lab3-primit/c/bool-char.c
#include <stdio.h>
#include <ctype.h>

int getNumber()
{
    char input[100];
    int result;
    int haveResult = 0; /* false */

    while (! haveResult)
    {
        printf("Input one number: ");
        fgets((char *)&input, 100, stdin); // get a string from user
        if ( 1 == sscanf(input, "%d", &result) ) // parse string as decimal integer
        {
            haveResult = 1; /* true */
        }
        else
        {
            printf("Invalid number, please try again.\n");
        }
    }
    return result;
}


int main(char** argv, int argc)
{
    int n = getNumber();

    while ( n = 0 ) // [nasty error]
    {
        printf("Please, input a non-zero number.\n");
        n = getNumber();
    }

    // [notice the conversion of int to a string]
    printf("You typed: %d\n", n);

    // [conditional expression example]
//     printf("n > 0 ? 100 : -100 = %d\n", n > 0 ? 100 : -100);

    // [check a function working with a char:]
//     printf("toupper(97) = '%c'\n", 97, toupper(97));

    return 0;
}
